recipe/bld.bat

```diff
@@ -18,4 +18,11 @@ cmake -G "NMake Makefiles"                     ^
 if errorlevel 1 exit /b 1
 cmake --build . --config Release --target INSTALL -- VERBOSE=1
 if errorlevel 1 exit /b 1
+
+REM For backward compatibility; previous build numbers provided libjpeg.dll/lib
+REM and then we changed it to jpeg.dll/lib to match upstream .pc contents
+REM and observed expectations in public projects
+copy %PREFIX%\Library\bin\jpeg.dll %PREFIX%\Library\bin\libjpeg.dll
+copy %PREFIX%\Library\lib\jpeg.lib %PREFIX%\Library\lib\libjpeg.lib
```

Comment:
Note that I think that the DLL name is baked into the import library (I think). From https://blog.aaronballman.com/2011/10/how-dll-imports-work/:

> When you create a Win32 project, you typically will see some additional dependencies listed in the linker section. These dependencies are for things like Kernel32, User32, et al. Those are not static libraries (which contain complete compiled binary data), but stub import libraries. Alternatively, you may tell the linker about a library to link against by using the #pragma comment(lib) directive. Either approach results in the same outcome — the linker knows about a stub library to import functions from. **The stub library includes information such as the DLL to locate (which explains why you cannot rename a DLL and have its functions located when the application launches)**, as well as the functions exported.

(emphasis mine)

At least that what I understand.

Reply:
JC is correct, the import library (`.lib`) found by the linker on its library search path from the `-lFOO` flag indicates which `.dll` to find the actual code in.  The DLL can be an unrelated filename, think passing `-lz` to the linker will use the `zlib1.dll` on MSYS2.

Try (something like):

```
dumpbin /symbols %PREFIX%\Library\lib\jpeg.lib | find "_head_"
```

if that reports (something like) `_head_jpeg_dll` then your `.lib` file is only good for `jpeg.dll` -- this is generally true but there are bound to be name-mangling issues which means it isn't true in some narrow cases.

You need to:

-  (re-)create a `.def` file which says `LIBRARY "libjpeg.dll"` (instead of `LIBRARY "jpeg.dll"` or no LIBRARY statement which implies the DLL uses the same base name) 
- you _can_ just copy the `.dll` to rename it
- rebuild the `.lib` import library

I don't have the specific commands as I type this.

Reply:
Ian's suggestions are the safe ones.
Currently, `imagecodecs` has issues with naming `jpeg.dll`/`libjbeg.dll`, see https://github.com/AnacondaRecipes/imagecodecs-feedstock/blob/master/recipe/meta.yaml#L14-L18

Reply:
I was thinking that generating it via CMake, i.e. keeping in the pre-existing `libjpeg` targets alongside the new `jpeg` ones, would be safest. Any thoughts?

Reply:
You would have a double compilation I think. In theory, you can remove line 25 and keep line 26. Copying `jpeg.lib` to `libjpeg.lib` will result in `libjpeg.lib` to point to `jpeg.dll`, so it _should_ word, but you would have to test to make sure that it really works. It's relatively easy to test buy just replacing `jpeg` with `libjpeg` in toarchvision for example.